metricflow/model/semantics/semantic_containers.py

```diff
@@ -291,22 +231,17 @@ def _add_data_source(
         fail_on_error: bool = True,
         logging_context: str = "",
     ) -> None:
-        """Add data source semantic information, validating consistency with existing data sources and establishing links"""
+        """Add data source semantic information, validating consistency with existing data sources."""
         errors = []
 
         if data_source.name in self._data_source_names:
             errors.append(f"name {data_source.name} already registered - please ensure data source names are unique")
 
-        if data_source.origin is DataSourceOrigin.DERIVED:
-            raise ValueError(
-                f"Cannot add derived data source (with name: {data_source.name}) to New DataSourceSemantics... for now"
-            )
-
         for measure in data_source.measures:
-            if measure.name in self._measure_aggs and self._measure_aggs[measure.name] != measure.agg:
+            if measure.reference in self._measure_aggs and self._measure_aggs[measure.reference] != measure.agg:
                 errors.append(
-                    f"conflicting aggregation (agg) for measure `{measure.name}` registered as `{self._measure_aggs[measure.name]}`; "
-                    f"Got `{measure.agg}"
+                    f"conflicting aggregation (agg) for measure `{measure.reference}` registered as "
```

Comment:
I think we probably want `measure.reference.element_name` here.

Reply:
Updated.